lib/if.c

```diff
@@ -216,6 +216,33 @@ void if_delete(struct interface *ifp)
 	XFREE(MTYPE_IF, ifp);
 }
 
+/* Build the table key */
+static void if_build_key(uint32_t ifindex, struct prefix *p)
+{
+	p->family = AF_INET;
+	p->prefixlen = IPV4_MAX_BITLEN;
+	p->u.prefix4.s_addr = ifindex;
+}
+
+/* Look up an interface by identifier within a NS */
+struct interface *if_lookup_by_index_per_route_table(
+						struct route_table *if_table,
+						uint32_t ifindex)
+{
```

Comment:
`if_lookup_by_index_per_route_table()` is not a good name, `if_lookup_by_index_per_ns()` was much better in terms of expressing what the function does.

Why did you have to move this function to _lib/_? I think only zebra is aware of namespaces. I think you should maintain `if_lookup_by_index_per_ns()` where it is right now.

Reply:
you are right. Idid not move that function.
I kept it on zebra